### DISCUSSION TITLE
Using digest in a stored procedure

### DISCUSSION THREAD

**Discussion by Tarang** [supabase/supabase#1096] [Questions] [answered]
Hi,

I am trying to use digest in a stored procedure
```
create or replace FUNCTION hello_world(
	)
	RETURNS text
	language plpgsql    
	as $$
	DECLARE

	BEGIN
		
			
		RETURN encode(digest(('HEY'::text)::bytea, 'sha1'), 'hex');
		
	END;
	$$
```

Then I try to call it using the supabase `rpc` method and i get this error
```
{"hint":"No function matches the given name and argument types. You might need to add explicit type casts.","details":null,"code":"42883","message":"function digest(bytea, unknown) does not exist"}
```

However, if I run it in SQL it works fine
<img width="655" alt="image" src="https://user-images.githubusercontent.com/1616628/113948174-a69bd180-984f-11eb-8bb0-c1a8384254af.png">

I'm not too sure whats going wrong here

**Comment by steve-chavez** [comment, marked as answer]
Try using it `digest` like:

```sql
select  encode(extensions.digest(('HEY'::text)::bytea, 'sha1'), 'hex');;
```

For more info, check: https://github.com/supabase/supabase/discussions/1048#discussion-3304631
